### PR TITLE
Update logger to use python logging

### PR DESCRIPTION
**Pull request by Sae86**
Signed-off-by: Sara Batllori <sara.batllori@intel.com>

### CODE REVIEW

**Comment by BrentHoltsclaw** [comment]
It appears the PR is removing the 'hal' logging level.  Can you explain why this is being removed.

File: chipsec/logger.py [hunk 1]
```diff
@@ -125,31 +125,76 @@ def getMessage(self) -> str:
...
-        # Respect https://no-color.org/ convention, and disable colorization
-        # when the output is not a terminal (eg. redirection to a file)
-        pyLogging.setLogRecordFactory(chipsecrecordfactory)  # applies colorization to output
-        self.rootLogger.addHandler(self.logstream)  # adds streamhandler to root logger
+        #---------------------------------------------------------------------------------------#
+        configPath = os.path.join(os.getcwd(), 'chipsec', 'logConfig', profile)
+        logPath = os.path.join(os.getcwd(), 'logs', strftime('%a%b%d%y-%H%M%S'))
+        logging.config.fileConfig(configPath, defaults={'filename': repr(logPath)[1:-1]})
+        logging.addLevelName(15, "VERBOSE")
+        logging.addLevelName(19, "verbose")
```
Comment (iherna2) [inline chipsec/logger.py:152]: What is the difference between 'VERBOSE' and 'verbose'

File: chipsec/logger.py [hunk 2]
```diff
@@ -125,31 +125,76 @@ def getMessage(self) -> str:
 class Logger:
     """Class for logging to console, text file, XML."""
 
-    def __init__(self):
+    class level(Enum):
+        DEBUG = 10
+        VERBOSE = 11
+        # HAL = 12
```
Comment (iherna2) [inline chipsec/logger.py:131]: Why are we removing hal?

File: chipsec/logger.py [hunk 3]
```diff
@@ -194,121 +239,135 @@ def disable(self):
...
 
-    def log(self, text):
-        """Plain Log message"""
-        self._log(text, pyLogging.INFO, "WHITE")
-
-    def log_verbose(self, text):
+    def log_verbose(self, text):  # Use log('text', self.logger.VERBOSE)
         """Logs a Verbose message"""
-        self._log(text, pyLogging.getLevelName("verbose"), "LIGHT_GRAY")
+        #self._log(text, logging.getLevelName("verbose"), "LIGHT_GRAY")
```
Comment (iherna2) [inline chipsec/logger.py:273]: please remove dead code

File: chipsec/logger.py [hunk 4]
```diff
@@ -507,13 +563,13 @@ def print_buffer_bytes(arr, length=16):
 
 def pretty_print_hex_buffer(arr, length=16):
     """Prints the buffer (bytes, bytearray) in a grid"""
-    _str = ["    _"]
+    _str = ['    _']
     for n in range(length):
-        _str += ["{:02X}__".format(n)]
+        _str += ['{:02X}__'.format(n)]
```
Comment (iherna2) [inline chipsec/logger.py:568]: shouldn't it be _str += [f'{n:02X}__'] for consistency?

File: chipsec/logger.py [hunk 5]
```diff
@@ -194,121 +247,135 @@ def disable(self):
...
-        self._log(text, pyLogging.getLevelName("verbose"), "LIGHT_GRAY")
+        #self._log(text, logging.getLevelName("verbose"), "LIGHT_GRAY")
+        self.log(text, self.level.VERBOSE)
 
-    def log_hal(self, text):
+    def log_hal(self, text):  # Use log("text", self.logger.VERBOSE)
         """Logs a hal message"""
-        self._log(text, pyLogging.getLevelName("hal"), "LIGHT_GRAY")
+        #self._log(text, logging.getLevelName("hal"), "LIGHT_GRAY")
+        self.log(text, self.level.VERBOSE)
```
Comment (BrentHoltsclaw) [inline chipsec/logger.py:287]: self.level.HAL?

File: chipsec/logger.py [hunk 6]
```diff
@@ -125,31 +125,84 @@ def getMessage(self) -> str:
...
+        INFO = 20
+        GOOD = 21
+        IMPORTANT = 22
+        WARNING = 30
+        BAD = 31
+        ERROR = 40
+        CRITICAL = 50
+        EXCEPTION = 60
+
+    def __init__(self, profile = 'debug.conf'):
```
Comment (BrentHoltsclaw) [inline chipsec/logger.py:142]: remove whitespace around equals parameter

File: chipsec/logger.py [hunk 7]
```diff
@@ -125,31 +125,84 @@ def getMessage(self) -> str:
...
-        # when the output is not a terminal (eg. redirection to a file)
-        pyLogging.setLogRecordFactory(chipsecrecordfactory)  # applies colorization to output
-        self.rootLogger.addHandler(self.logstream)  # adds streamhandler to root logger
+        self.logstream = logging.StreamHandler(sys.stdout)
+        logging.addLevelName(19, "verbose")
+        logging.addLevelName(18, "hal")
+        #---------------------------------------------------------------------------------------#
+        configPath = os.path.join(os.getcwd(), 'chipsec', 'logConfig', profile)
+        logPath = os.path.join(os.getcwd(), 'logs', strftime('%a%b%d%y-%H%M%S'))
+        logging.config.fileConfig(configPath, defaults={'filename': repr(logPath)[1:-1]})
```
Comment (BrentHoltsclaw) [inline chipsec/logger.py:153]: looks like chipsec is missing the logs directory may want to check if it exists and create if it doesn't

File: chipsec/logger.py [hunk 8]
```diff
@@ -194,121 +247,135 @@ def disable(self):
...
         except BaseException:
             print(text)
 
-    def log(self, text):
-        """Plain Log message"""
-        self._log(text, pyLogging.INFO, "WHITE")
-
-    def log_verbose(self, text):
+    def log_verbose(self, text):  # Use log('text', self.logger.VERBOSE)
         """Logs a Verbose message"""
```
Comment (BrentHoltsclaw) [inline chipsec/logger.py:280]: Can log_hal, log_verbose, and log_debug be left off the deprecated function list.  I see value in calling them directly.

Comment (Sae86) [reply chipsec/logger.py:280]: The idea is to only have one log function. Do you not see the same value between log_hal() and log(level.HAL)? What makes those better options?

Comment (Sae86) [reply chipsec/logger.py:279]: Exactly. We can discuss details later

File: chipsec/logger.py (hunk 1 above)
Comment (Sae86) [reply chipsec/logger.py]: None other than case, since it is case sensitive. Updated to just use Upper case

File: chipsec/logger.py (hunk 2 above)
Comment (Sae86) [reply chipsec/logger.py]: Not removing it, it was a placeholder. It's now been added.

File: chipsec/logger.py [hunk 9]
```diff
@@ -345,12 +374,12 @@ def print_banner(self, arguments: Sequence[str]) -> None:
         self.log(f'[CHIPSEC] Arguments: {args}')
         self.log(message)
 
-    def print_banner_properties(self, cs) -> None:
+    def print_banner_properties(self, cs, os_version) -> None:
```
Comment (BrentHoltsclaw) [inline chipsec/logger.py:377]: can this be changed to pass in the helper string from calling cs.helper.helper.get_info() opposed to expecting a cs object?

File: chipsec/logger.py (hunk 8 above)
Comment (BrentHoltsclaw) [reply chipsec/logger.py:277]: When would be a good time to discuss?

**Review by frinzell — APPROVED** [review]

File: chipsec/logger.py [hunk 10]
```diff
@@ -70,6 +71,7 @@ class chipsecrecordfactory(pyLogging.LogRecord):
             }
 
             def getMessage(self) -> str:
+                if self.name != "CHIPSEC": return super().getMessage()
```
Comment (npmitche) [inline chipsec/logger.py:74]: See comment below: re "CHIPSEC" string

Comment (npmitche) [reply chipsec/logger.py:74]: ... comment above... depending on how/where you read this.

File: chipsec/logger.py (hunk 8 above)
Comment (Sae86) [reply chipsec/logger.py:278]: let's set up a meeting

**Review by dscott90 — APPROVED** [review]

**Review by npmitche — APPROVED** [review]

File: chipsec/logger.py (hunk 9 above)
Comment (Sae86) [reply chipsec/logger.py:379]: will look into this for next release

**Pull request merged**